lua/toggleterm/terminal.lua

```diff
@@ -360,6 +360,19 @@ local function opener(size, term)
   end
 end
 
+---Spawn terminal background job in a buffer without a window
+function Terminal:spawn()
+  if fn.bufexists(self.bufnr) == 0 then
+    self.bufnr = vim.api.nvim_create_buf(false, false)
```

Comment:
So looking a bit at this now, UI code show generally be in the `ui` module, i.e. all window/buffer related stuff should live in there e.g. `ui.create_buf({ open = false})` or something.

Reply:
This is basically the same code as in [Terminal:open()](https://github.com/akinsho/toggleterm.nvim/blob/6c7f5dbdd69bc5611a85194ddca83ac2c8ee84d6/lua/toggleterm/terminal.lua#L373-L381), with the addition of `nvim_create_buf`, so shouldn't it rather stay nearby? Or do you mean refactoring both?

Reply:
Sorry I just meant the function call that creates a buffer should use the UI modules instead so there is a strict separation of concerns. The spawn function is fine where it is

Reply:
I moved buffer creation to ui module with: https://github.com/akinsho/toggleterm.nvim/pull/218/commits/dcc2660fcb48eb989066c91a33d365dead45911c